readiness/controller/pkg/readinessprovider/suite_test.go

```diff
@@ -379,16 +390,123 @@ var _ = Describe("Readiness Provider controller", func() {
 		err = k8sClient.Delete(ctx, readinessProvider)
 		Expect(err).To(BeNil())
 	})
+
+	It("should fail when service account details are partial", func() {
+		readinessProvider := getTestReadinessProvider()
+		readinessProvider.Spec.ServiceAccount = &corev1alpha2.ServiceAccountSource{
+			Name:      "default",
+			Namespace: "",
+		}
+		err := k8sClient.Create(ctx, readinessProvider)
+		Expect(err).NotTo(BeNil())
+	})
+
+	It("should fail when service account does not exist", func() {
+		readinessProvider := getTestReadinessProvider()
+		readinessProvider.Spec.ServiceAccount = &corev1alpha2.ServiceAccountSource{
+			Name:      "non-existent-sa",
+			Namespace: "default",
+		}
+		err := k8sClient.Create(ctx, readinessProvider)
+		Expect(err).NotTo(BeNil())
+	})
+
+	It("should succeed when service account has required roles", func() {
+		newPod := corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testpod",
+				Namespace: "default",
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "test-container",
+						Image: "test:tag",
+					},
+				},
+			},
+		}
+
+		err := k8sClient.Create(context.TODO(), &newPod)
+		Expect(err).To(BeNil())
+
+		readinessProvider := getTestReadinessProvider()
+		readinessProvider.Spec.ServiceAccount = &corev1alpha2.ServiceAccountSource{
+			Name:      "pod-sa",
+			Namespace: "default",
+		}
+		readinessProvider.Spec.Conditions = append(readinessProvider.Spec.Conditions, corev1alpha2.ReadinessProviderCondition{
+			Name: "pod-exists",
+			ResourceExistenceCondition: &corev1alpha2.ResourceExistenceCondition{
```

Comment:
This does not call the actual resource existence condition.

Reply:
I think this test case should move to the conditions package.

Reply:
Done